src/test/java/com/onlinebookstore/service/impl/ShoppingCartServiceImplTest.java

```diff
@@ -0,0 +1,334 @@
+package com.onlinebookstore.service.impl;
+
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.onlinebookstore.dto.cartitem.CartItemRequestDto;
+import com.onlinebookstore.dto.cartitem.CartItemUpdateDto;
+import com.onlinebookstore.dto.shoppingcart.ShoppingCartResponseDto;
+import com.onlinebookstore.exception.EntityNotFoundException;
+import com.onlinebookstore.mapper.CartItemMapper;
+import com.onlinebookstore.mapper.ShoppingCartMapper;
+import com.onlinebookstore.model.Book;
+import com.onlinebookstore.model.CartItem;
+import com.onlinebookstore.model.ShoppingCart;
+import com.onlinebookstore.model.User;
+import com.onlinebookstore.repository.CartItemRepository;
+import com.onlinebookstore.repository.ShoppingCartRepository;
+import com.onlinebookstore.repository.UserRepository;
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.Optional;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@ContextConfiguration(classes = {ShoppingCartServiceImpl.class})
+@RunWith(SpringJUnit4ClassRunner.class)
+public class ShoppingCartServiceImplTest {
+    @MockBean
+    private CartItemMapper cartItemMapper;
+
+    @MockBean
+    private CartItemRepository cartItemRepository;
+
+    @MockBean
+    private ShoppingCartMapper shoppingCartMapper;
+
+    @MockBean
+    private ShoppingCartRepository shoppingCartRepository;
+
+    @Autowired
+    private ShoppingCartServiceImpl shoppingCartServiceImpl;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @Test
+    public void testGetShoppingCartDtoByUserId() {
+        User user = new User();
+        user.setDeleted(true);
+        user.setEmail("jane.doe@example.org");
+        user.setFirstName("Jane");
+        user.setId(1L);
+        user.setLastName("Doe");
```

Comment:
You do many extra work. See how we can test Book for example
![image](https://github.com/andrey-kiyas/OnlineBookStore/assets/20326858/2bcb3627-c2af-4c15-b50c-e99c3f597a5d)